agent/agent.go

```diff
@@ -300,10 +301,12 @@ func (a *agent) createTailnet(ctx context.Context, derpMap *tailcfg.DERPMap) (_
 		}
 	}()
 	if err = a.trackConnGoroutine(func() {
+		logger := a.logger.Named("reconnecting-pty")
```

Comment:
Review: Named logging seemed like it would be helpful here, even though it's not fully a separate service.